wit/lock.wit

```diff
@@ -0,0 +1,8 @@
+/// locker interface provides the ability to acquire locks
+interface locker {
+    use types.{milliseconds, error, lock};
+
+    /// acquire will attempt to obtain a lease for a given string key for a time to live (ttl)
+    /// in milliseconds.
+    acquire: func(key: string, ttl: milliseconds) -> result<lock, error>;
```

Comment:
What's your thoughts on acquiring a distributed lock and getting a lock and a **fencing token** back? 

fencing token is a way to ensure a write to the persistent service that the lock may be dealyed. See "Making the lock safe with fencing" in [How to Do Disitrbuted Locking](https://martin.kleppmann.com/2016/02/08/how-to-do-distributed-locking.html)

Reply:
+1 on encouraging fencing tokens - I'm not sure everyone has support for them (also for cases like redis, where gettimeofday is used for expiration tracking, rather than a monotonic clock) so I'm not sure how we could generically expose them :(